### PR TITLE
Create EventGrid Track 2 release pipeline

### PR DESCRIPTION
**Pull request by Soreloser2**
Edit ci.yaml to include release pipeline for Track 2 Event Grid library

### CODE REVIEW

**Review by samvaity — APPROVED** [review]

File: sdk/eventgrid/ci.yml [hunk 1]
```diff
@@ -44,3 +44,6 @@ extends:
       - name: azure-eventgrid
```
Comment (samvaity) [inline sdk/eventgrid/ci.yml:44]: Just confirming, so the same pipeline will be used for track1 and track2 releases?

Comment (Soreloser2) [reply sdk/eventgrid/ci.yml:44]: I believe that's how it would work, but I think we wouldn't approve the Track 1 package when we release.

**Comment by danieljurek** [comment]
Looking at other examples around the repo I think we ship track 1 from `ci.data.yml` and track 2 from `ci.yml`: 

Examples: 

Storage: 
 
* Track 2 (ci.yml) -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/ci.yml 
* Track 1 (ci.data.yml) -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/ci.data.yml

Key Vault: 

* Track 2 -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/keyvault/ci.yml
* Track 1 -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/keyvault/ci.data.yml

Service Bus: 

* Track 2 -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/ci.yml
* Track 1 -- https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/ci.data.yml

To that end I think these should be split so that Track 1 ships from a `ci.data.yml` and Track 2 ships from `ci.yml`

File: sdk/eventgrid/ci.yml (hunk 1 above)
Comment (Soreloser2) [reply sdk/eventgrid/ci.yml]: The new changes should mean that the pipelines are now separate

File: sdk/eventgrid/ci.yml [hunk 2]
```diff
@@ -35,12 +36,13 @@ pr:
       - sdk/eventgrid/mgmt-v2019_06_01
       - sdk/eventgrid/mgmt-v2020_01_01_preview
       - sdk/eventgrid/mgmt-v2020_04_01_preview
+      - sdk/eventgrid/microsoft-azure-eventgrid
 
 extends:
   template: ../../eng/pipelines/templates/stages/archetype-sdk-data.yml
```
Comment (alzimmermsft) [inline sdk/eventgrid/ci.yml:42]: You'll want to use `archetype-sdk-client.yml` as the template.

File: sdk/eventgrid/ci.data.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+trigger:
```
Comment (alzimmermsft) [inline sdk/eventgrid/ci.data.yml:2]: Indentation levels for this need to be 2, looks like it is 4 right now

**Comment by Soreloser2** [comment]
The build from source seems to be failing in different places

File: sdk/eventgrid/tests.yml [hunk 4]
```diff
@@ -11,4 +11,4 @@ jobs:
       Artifacts:
```
Comment (samvaity) [inline sdk/eventgrid/tests.yml:11]: Should this be 
>        AZURE_EVENTGRID_TEST_MODE: LIVE
changed to
> AZURE_TEST_MODE: RECORD

**Review by weshaggard — APPROVED** [review]

**Comment by weshaggard** [comment]
check-enforcer is having issues right now so I forced in the merge.

**Pull request merged**